cirq-core/cirq/circuits/circuit.py

```diff
@@ -1712,12 +1712,26 @@ def __init__(
                 together. This option does not affect later insertions into the
                 circuit.
         """
-        self._moments: List['cirq.Moment'] = []
-        with _compat.block_overlapping_deprecation('.*'):
-            if strategy == InsertStrategy.EARLIEST:
-                self._load_contents_with_earliest_strategy(contents)
-            else:
-                self.append(contents, strategy=strategy)
+        self._moments: List[Moment] = []
+        if len(contents) == 1 and isinstance(contents[0], AbstractCircuit):
+            self._moments.extend(contents[0].moments)
```

Comment:
This could produce two circuits with shared Moments.
Are Moment immutable?  If so this should be OK...

Reply:
Yes, moments are immutable.